### DISCUSSION TITLE
C

### DISCUSSION THREAD

**Discussion by srem1** [markqvist/Reticulum#217] [General]
I need a library in C.

Whatever we write must be in C. Whether we make a port to Ruby or embedded in a web server or whatever else needs C.

Preferably a client. Can be minimal. But please let's start doing it. You can just make a module for python to start with. The module in C will slowly transfer all the functionality from python to C. But C is very much needed.

**Comment by markqvist** [comment]
We all need a library in C dude ;)

It's coming. As soon as the Python reference implementation is complete, it will be one of my highest priorities.

**Comment by fab1an2** [comment]
Where would you like to start? What do you need to move first? Something small. To make it happen. :)

And next we will create more difficult things.
Provide some code. Maybe this library in Python is already written in C?

**Comment by wir19** [comment]
If that's important to you, then he suggests just embeded python in a C library.
Simple easy and You will make a test and program in C with this lib.
Slow?Big exec? Yes, but pieces to pieces you can remove element from python to C.
All component arel still working, movint to strict  -std=c99 -Wall -pedantic -pedantic-errors ;) will be easy.
